### ISSUE TITLE
Cannot Install tensorflow-gpu on python=3.10.11

### ISSUE THREAD

**Issue by shvamabps** [tensorflow/tensorflow#60243]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Build/Install

### Have you reproduced the bug with TF nightly?

No

### Source

source

### Tensorflow Version

2.12.0

### Custom Code

No

### OS Platform and Distribution

Windows 11

### Python version

3.10.11

### Current Behaviour?

```shell
A bug happened!
```

### Standalone code to reproduce the issue

```shell
pip install tensorflow-gpu
```

### Relevant log output

```shell
Collecting tensorflow-gpu
  Downloading tensorflow-gpu-2.12.0.tar.gz (2.6 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [41 lines of output]
      Traceback (most recent call last):
        File "C:\Users\shiva\OneDrive\Desktop\LipNet\venv\lib\site-packages\setuptools\_vendor\packaging\requirements.py", line 35, in __init__  
          parsed = parse_requirement(requirement_string)
        File "C:\Users\shiva\OneDrive\Desktop\LipNet\venv\lib\site-packages\setuptools\_vendor\packaging\_parser.py", line 64, in parse_requirement
          return _parse_requirement(Tokenizer(source, rules=DEFAULT_RULES))
        File "C:\Users\shiva\OneDrive\Desktop\LipNet\venv\lib\site-packages\setuptools\_vendor\packaging\_parser.py", line 82, in _parse_requirement
          url, specifier, marker = _parse_requirement_details(tokenizer)
        File "C:\Users\shiva\OneDrive\Desktop\LipNet\venv\lib\site-packages\setuptools\_vendor\packaging\_parser.py", line 126, in _parse_requirement_details
          marker = _parse_requirement_marker(
        File "C:\Users\shiva\OneDrive\Desktop\LipNet\venv\lib\site-packages\setuptools\_vendor\packaging\_parser.py", line 147, in _parse_requirement_marker
          tokenizer.raise_syntax_error(
        File "C:\Users\shiva\OneDrive\Desktop\LipNet\venv\lib\site-packages\setuptools\_vendor\packaging\_tokenizer.py", line 163, in raise_syntax_error
          raise ParserSyntaxError(
      setuptools.extern.packaging._tokenizer.ParserSyntaxError: Expected end or semicolon (after name and no valid version specifier)
          python_version>"3.7"
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```
</details>

**Comment by KrishnaSChavan** [comment]
TensorFlow GPU is currently not available for Python 3.10.11
Downgrade your Python version to one of the supported versions 
Just uninstall current python and reinstall python version like  3.9  or below to use gpu.

**Comment by shvamabps** [comment]
@KrishnaSChavan  I installed 3.9.12 version, still it wasn't working.
According to you, which python version is supported for tf-gpu.

**Comment by mihaimaruseac** [comment]
Please use `tensorflow` instead of `tensorflow-gpu`. See first bullet in TF 2.12 release notes: https://github.com/tensorflow/tensorflow/releases/tag/v2.12.0

**Comment by sushreebarsa** [comment]
@shvamabps Please refer to the comment above and let us know if that helps?
Thank you!

**Comment by shvamabps** [comment]
![image](https://user-images.githubusercontent.com/37547983/230294789-2736964d-7bce-42ce-909a-3b8371813765.png)
Actually, I am using conda, the above image shows more details regarding tf.
I have got GPU (Nvidia GeForce RTX 3050) but it is not recognizing it.

**Comment by sushreebarsa** [comment]
@shvamabps For the users of TF v2.11 or above are recommended to use Windows WSL for GPU Access. Could you please see the instructions here - https://www.tensorflow.org/install/pip#windows-wsl2
If you are using Conda, you can create a Conda environment and install inside the conda env.
You could also give docker containers a try if you prefer that over creating your own conda env (https://www.tensorflow.org/install/docker)
Thank you!

**Comment by shvamabps** [comment]
I have tried all of that. Torch recognizes the GPU but tf doesn't

**Comment by SuryanarayanaY** [comment]
Hi @shvamabps ,

Please cross check the detailed instructions mentioned in documentation for [windows-native](https://www.tensorflow.org/install/pip#step-by-step_instructions) and for [wsl2](https://www.tensorflow.org/install/pip#step-by-step_instructions) for enabling GPU setup. By default Tensorflow will not detect GPU unless you install GPU driver and then CUDA/cuDNN toolkit and setting the path and all these steps are manual.

After ensuring installation of GPU driver ensure it is up and running with `nvidia-smi` command. Installation instructions of GPU driver is not explicitly mentioned in Tensorflow documentation. You can check with nvidia website for same. If you have already installed GPU driver please share the output of `nvidia-smi` .

Please use `pip install tensorflow` only instead of `tensorflow-gpu`. Please make a note again that TF <=2.10 version you can have GPU support on native windows however for TF>=2.11 you should install wsl2 on windows to use GPU support.

Thanks!

**Comment by shvamabps** [comment]
Why do I have to install in WSL2. Can't the only windows not be used for that purpose?? What issues are there in that case if I use TF without WSL2.

**Comment by mihaimaruseac** [comment]
Please see https://github.com/tensorflow/tensorflow/issues/59918#issuecomment-1513386150 Due to the fact that there are so few developers with Windows expertise in TF OSS team and so few users of TF GPU on Windows it was decided more than a year ago to drop native GPU support on Windows and instead use the Linux support provided by WSL2. There is also the alternative of using a Linux environment, cloud, Google Collab, etc.

**Comment by SuryanarayanaY** [comment]
Hi @shvamabps ,

Please have a look of above [comment](https://github.com/tensorflow/tensorflow/issues/60243#issuecomment-1513563671). WSL2 is needed on windows to enable GPU to be utilized by TF(>=2.11 version) and the purpose is ease of maintenance of builds by our team and also GPU performance on native windows might not be as good as compared to Linux (WSL2) environment . More context can be seen in attached issue in above comment.

**Comment by shvamabps** [comment]
Thanks for the clarifications. 
It was a lot helpful.

**Comment by SuryanarayanaY** [comment]
Hi @shvamabps ,

Glad that we are of some help. Could you please spare some time to close the issue if resolved. Thanks

**Comment by boxabirds** [comment]
I'm experiencing the exact same error under ubuntu 22.04 with CUDA 12.1. And as per other commenters, installing just `tf-nightly` installs ok but prevents my setup from using my RTX 4090 GPU at all which is… not ideal.

The same setup has Pytorch running fine so it's not an issue with CUDA or anything else.

**Comment by dfrankow** [comment]
I had the same problem with python 3.11 on OS X. I don't think I have a GPU. I commented out the `tensorflow-gpu` lines in `requirements.txt`, and everything installed.

Now maybe nothing will work well without a GPU, we'll see.

**Comment by mnwato** [comment]
same problem with `python==3.10.6` and `tensorflow==2.12.0` on ubuntu 20.04.
I have checked colab python and tensorflow version: `python==3.10.12`  , `tensorflow=2.12.0`
Any help is appreciated.

**Comment by ioneliabuzatu** [comment]
why is this closed? the problem is still on

**Comment by boxabirds** [comment]
Because tensorflow-gpu has been deprecated.

**Comment by ioneliabuzatu** [comment]
ah ok. 

it seems like there is no release of tensorflow for cuda 12 yet. i got the same error on python37 and above.

**Comment by bestwebdev0728** [comment]
I already installed a python 3.11.6 and cuda12.3 and cudnn 8.9.6. And I install tensorflow[and-cuda]. But I cannot find nvidia. So I can't use nvidia. What can I do to solve this problem. Additionaly Could you tell me the versions of nvidia driver, cuda, cudnn and tensorflow to use nvidia?

**Comment by goodevile** [comment]
answering the question, tensorflow-gpu will not work sometimes, you will have to know what you are doing, first check the compatibility of:

 TensorFlow + Python + Nvidia "stuff" (drivers, CUDA toolkit, cuDNN)

If any of these are incompatible with each other for what you are doing then you need to check what you have and install the proper versions for each.

I am telling you this by assuming that your WSL2 is working properly and you understand how it works, if not then watch some videos that some folks explain how it works for you to understand if you can do what you want on WSL2 or try Ubuntu from the start.

FYI: [https://www.tensorflow.org/install/source_windows](url)

> Note: GPU support on native-Windows is only available for 2.10 or earlier versions, starting in TF 2.11, CUDA build is not supported for Windows. For using TensorFlow GPU on Windows, you will need to build/install TensorFlow in WSL2 or use tensorflow-cpu with TensorFlow-DirectML-Plugin


Advice: 

-I strongly recommend using Ubuntu, WSL2 is good but Ubuntu is simply the best for the resources it consumes when compared to Windows.
-Start asking yourself which version of Tensorflow you need then install all dependencies for GPU support.
